web/src/pm_list.ts

```diff
@@ -144,6 +156,18 @@ function scroll_all_private_into_view(): void {
 
 export function handle_narrow_activated(filter: Filter): void {
     const active_filter = filter;
+    is_private_messages_in_view = active_filter.has_operator("dm");
+    unhighlight_dm_list_item();
+    if (!is_private_messages_in_view) {
+        /*
+            Since handle_narrow_activated is called within
+            narrow.activate, and narrow.activate is responsible
+            for building narrows, this is the most efficient place
+            to handle user navigation away from a DM view.
+        */
+        close();
```

Comment:
Is this necessary since it will change the current design by collapsing DM every time user navigates away. Maybe we could just use `update_private_messages` directly

Reply:
That behavior is definitely undesirable.

Reply:
Hi @alya, @pratik-pc 

Thank you both for taking the time to review my pull request. I appreciate your feedback.

However, I'd like to clarify that the primary purpose of this PR is to close the Direct Messages view when the user navigates away. As per the comment from Tim [here](https://github.com/zulip/zulip/pull/28325#issuecomment-1897561567)

Reply:
#27698 does not request any changes to when the DMs section collapses, which is consistent with @timabbott 's comment.